pkg/manifests/init.go

```diff
@@ -0,0 +1,203 @@
+package manifests
+
+import (
+	"bytes"
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"html/template"
+	"os"
+	"path"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/loft-sh/vcluster/pkg/constants"
+	"github.com/loft-sh/vcluster/pkg/util"
+	"github.com/loft-sh/vcluster/pkg/util/applier"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	InitManifestRelativePath = "init/initmanifests.yaml"
+	DEFAULT_NAMESPACE        = "default"
+)
+
+var (
+	LAST_APPLIED_MANIFEST_HASH string
+
+	lastAppliedObjects util.UnstructuredMap
+	ErrorEmptyManifest = errors.New("empty manifest file")
+)
+
+func init() {
+	lastAppliedObjects = make(util.UnstructuredMap)
```

Comment:
Please do not save this globally here and instead save it in a configmap inside the virtual cluster. Rather handle this in the controller though instead of this helper file.